kalite/topic_tools/__init__.py

```diff
@@ -81,26 +82,32 @@ def recurse_nodes(node):
             else:
                 # By default this is very charitable, assuming if something has not been annotated
                 # it is available - needs to be updated for exercises.
-                if content_cache.get(node.get("id"), {}).get("languages", True):
+                if content_cache.get(node.get("id"), {}).get("available", True):
                     node["available"] = True
 
-        recurse_nodes(TOPICS[channel])
+            # Translate everything for good measure
+            translation.activate(i18n.lcode_to_django_lang(language))
```

Comment:
Hrm, not sure I understand. The problem I had is that doing this at the whole function level would cause problems, as when the individual content items get translated, the translation gets switched to the best match for them instead (which could mean that the topics translation now switches over to a different language, because the content did not have translation available).